lib/ufolint/validators/plistvalidators.py

```diff
@@ -289,23 +290,44 @@ def run_ufolib_import_validation(self):
         ss = StdStreamer(self.ufopath)
         for glyphs_dir in self.ufoobj.glyphsdir_list:
             res = Result(glyphs_dir[1])
-            rel_dir_path = os.path.join(self.ufopath, glyphs_dir[1])
+            rel_dir_path = Path(self.ufopath, glyphs_dir[1])
             try:
                 # read contents.plist with ufoLib as GlyphSet instantiation
                 # the ufoLib library performs type validations on values on read
                 # glyphs_dir_list is a list of lists mapped to glyphs dir name,
                 # glyphs dir path
-                GlyphSet(
+                gs = GlyphSet(
                     rel_dir_path, ufoFormatVersion=self.ufoversion, validateRead=True
                 )  # test for raised exceptions
                 res.test_failed = False
+
+                # Check for unlisted files if contents.plist exists. It is mandated by
+                # the spec, but if it does not exist, glifLib will shrug and say the
+                # glyph set was empty.
+                if (rel_dir_path / self.testfile).exists():
+                    files = set(gs.contents.values())
+                    files.update(("contents.plist", "layerinfo.plist"))
+                    files_actually = set(
+                        str(p.relative_to(rel_dir_path))
+                        for p in rel_dir_path.glob("**/*")
+                    )
+                    unlisted_files = files_actually - files
+                    if unlisted_files:
+                        res.test_failed = True
+                        res.exit_failure = True  # sacrilege
+                        res.test_long_stdstream_string = (
+                            f"{str(rel_dir_path)} contains rogue files not listed in contents.plist: "
+                            f"{', '.join(unlisted_files)}"
+                        )
+                        self.test_fail_list.append(res)
+
                 ss.stream_result(res)
             except Exception as e:
                 res.test_failed = True
                 res.exit_failure = True  # mandatory file
                 res.test_long_stdstream_string = (
                     "contents.plist in "
-                    + rel_dir_path
+                    + str(rel_dir_path)
```

Comment:
I'm not opposed to f-strings if we want to begin to make the conversion.  This was pre-fonttools Py3.6+ only support era source.

Reply:
This was just a quick make-it-work change, f-strings conversion should be a separate PR imho.